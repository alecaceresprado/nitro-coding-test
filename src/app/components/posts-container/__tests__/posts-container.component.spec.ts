import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import {
  PostsContainerComponent,
  GoupingSelectorComponent,
  PostsTreeComponent,
  PostDetailsComponent,
} from '../../../components';
import { OrderCriteria } from '@models';

const mockDb = require('../../../../../db.json');

describe('PostsContainerComponent', () => {
  let component: PostsContainerComponent;
  let fixture: ComponentFixture<PostsContainerComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        PostsContainerComponent,
        GoupingSelectorComponent,
        PostsTreeComponent,
        PostDetailsComponent,
      ],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(PostsContainerComponent);
    component = fixture.componentInstance;
    component.posts = mockDb.posts;
    component.ngOnChanges();
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should render', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should setup groupings correctly', () => {
    expect(component.groupings).toMatchSnapshot();
  });

  it('should change criteria and accumulate groupings', () => {
    component.handleCriteriaChange(OrderCriteria.LOCATION);
    fixture.detectChanges();
    expect(component.groupings).toMatchSnapshot();
  });

  it('should handle postClick correctly', () => {
    component.handlePostClick(3);
    fixture.detectChanges();
    expect(component.selectedPostId).toEqual(3);
  });
});
