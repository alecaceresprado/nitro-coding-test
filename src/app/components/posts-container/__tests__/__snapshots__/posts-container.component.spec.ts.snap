// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsContainerComponent should change criteria and accumulate groupings 1`] = `
Object {
  "By location": Object {
    "Dublin": Array [
      Object {
        "author": "Happy Manager",
        "id": 5,
        "location": "Dublin",
        "text": "A modern PDF annotator that canaccommodate all of the cooks in a very busy kitchen is what your employees really need.",
        "time": "1553080742",
      },
      Object {
        "author": "Happy Manager",
        "id": 6,
        "location": "Dublin",
        "text": "An integratedproductivity solution breaks information through barriers and allows workers to collaborate in real time.",
        "time": "1553099742",
      },
    ],
    "San Francisco": Array [
      Object {
        "author": "Happy User",
        "id": 1,
        "location": "San Francisco",
        "text": "Proper PDF conversion ensures that every element of your document remains just as you left it.",
        "time": "1591699872486",
      },
      Object {
        "author": "Happy User",
        "id": 2,
        "location": "San Francisco",
        "text": "The modern workplace is increasingly digital, and workflows are constantly evolving. ",
        "time": "1501690872486",
      },
      Object {
        "author": "Happy Developer",
        "id": 3,
        "location": "San Francisco",
        "text": "Digital transformation isn’t justa buzzword",
        "time": "1501690872999",
      },
    ],
    "Sydney": Array [
      Object {
        "author": "Happy Developer",
        "id": 4,
        "location": "Sydney",
        "text": "An expectation of digital efficiency has become the norm in our daily lives",
        "time": "1591699872999",
      },
    ],
  },
  "By posting week": Object {
    "Week 24": Array [
      Object {
        "author": "Happy User",
        "id": 1,
        "location": "San Francisco",
        "text": "Proper PDF conversion ensures that every element of your document remains just as you left it.",
        "time": "1591699872486",
      },
      Object {
        "author": "Happy Developer",
        "id": 4,
        "location": "Sydney",
        "text": "An expectation of digital efficiency has become the norm in our daily lives",
        "time": "1591699872999",
      },
    ],
    "Week 31": Array [
      Object {
        "author": "Happy User",
        "id": 2,
        "location": "San Francisco",
        "text": "The modern workplace is increasingly digital, and workflows are constantly evolving. ",
        "time": "1501690872486",
      },
      Object {
        "author": "Happy Developer",
        "id": 3,
        "location": "San Francisco",
        "text": "Digital transformation isn’t justa buzzword",
        "time": "1501690872999",
      },
    ],
    "Week 4": Array [
      Object {
        "author": "Happy Manager",
        "id": 5,
        "location": "Dublin",
        "text": "A modern PDF annotator that canaccommodate all of the cooks in a very busy kitchen is what your employees really need.",
        "time": "1553080742",
      },
      Object {
        "author": "Happy Manager",
        "id": 6,
        "location": "Dublin",
        "text": "An integratedproductivity solution breaks information through barriers and allows workers to collaborate in real time.",
        "time": "1553099742",
      },
    ],
  },
}
`;

exports[`PostsContainerComponent should render 1`] = `
<app-posts-container
  groupings={[Function Object]}
  postToShow="undefined"
  posts={[Function Array]}
  selectedCriteria={[Function String]}
>
  <div
    class="posts"
  >
    <app-posts-tree
      class="posts__tree"
    >
      <div
        class="posts-list"
      >
        <div
          class="posts-list__title"
        >
          Posts
        </div>
        
        <div
          class="posts-list__parent-level"
        >
           Week 4 
          
        </div>
        <div
          class="posts-list__parent-level"
        >
           Week 24 
          
        </div>
        <div
          class="posts-list__parent-level"
        >
           Week 31 
          
        </div>
      </div>
    </app-posts-tree>
    <app-post-details
      class="posts__display"
    >
      <div
        class="post-details"
      >
        <div
          class="post-details__title"
        >
          Post details
        </div>
        
        <div
          class="post-details__header"
        >
           Please select a post from the list on your left 
        </div>
      </div>
      
    </app-post-details>
    <app-gouping-selector
      class="posts__criteria"
    >
      <div
        class="grouping-selector"
      >
        <div
          class="grouping-selector__title"
        >
          Options
        </div>
        <div
          class="grouping-selector__criteria"
        >
           Order Criteria 
          
          <p
            class="grouping-selector__criteria-element grouping-selector__criteria-element--selected"
          >
             By posting week 
          </p>
          <p
            class="grouping-selector__criteria-element"
          >
             By location 
          </p>
          <p
            class="grouping-selector__criteria-element"
          >
             By author 
          </p>
        </div>
      </div>
    </app-gouping-selector>
  </div>
</app-posts-container>
`;

exports[`PostsContainerComponent should setup groupings correctly 1`] = `
Object {
  "By posting week": Object {
    "Week 24": Array [
      Object {
        "author": "Happy User",
        "id": 1,
        "location": "San Francisco",
        "text": "Proper PDF conversion ensures that every element of your document remains just as you left it.",
        "time": "1591699872486",
      },
      Object {
        "author": "Happy Developer",
        "id": 4,
        "location": "Sydney",
        "text": "An expectation of digital efficiency has become the norm in our daily lives",
        "time": "1591699872999",
      },
    ],
    "Week 31": Array [
      Object {
        "author": "Happy User",
        "id": 2,
        "location": "San Francisco",
        "text": "The modern workplace is increasingly digital, and workflows are constantly evolving. ",
        "time": "1501690872486",
      },
      Object {
        "author": "Happy Developer",
        "id": 3,
        "location": "San Francisco",
        "text": "Digital transformation isn’t justa buzzword",
        "time": "1501690872999",
      },
    ],
    "Week 4": Array [
      Object {
        "author": "Happy Manager",
        "id": 5,
        "location": "Dublin",
        "text": "A modern PDF annotator that canaccommodate all of the cooks in a very busy kitchen is what your employees really need.",
        "time": "1553080742",
      },
      Object {
        "author": "Happy Manager",
        "id": 6,
        "location": "Dublin",
        "text": "An integratedproductivity solution breaks information through barriers and allows workers to collaborate in real time.",
        "time": "1553099742",
      },
    ],
  },
}
`;
